feedhenry/fh-messaging
# fix(Unable to find IP geolocation): Upgrade version of the geoip-lite from ~1.0.5 to  1.3.1

# Jira link(s)
https://issues.jboss.org/browse/RHMAP-21092

# What
Upgrade version of the geoip-lite from ~1.0.5 to 1.3.1

# Why
Intermittent errors as follows are faced in the logs.

```
{"name":"messaging","hostname":"ema1-core-util1","pid":11303,"level":40,"msg":"Error from geoip lookup, continuing anyway with no country for ip: 107.178.192.64","time":"2018-03-13T11:31:00.974Z","v":0}
```
 
This lib is responsible to get the geolocation of the IP received the requests of the client applications for metrics proposes. Then, since we are with an old version and the latest did not have break changes but contain fix bugs and improvements related to this action the best approach here in order to try to solve or at least avoid this issue is using it.  

# How
`npm install --save -save-exact geoip-lite@1.3.1`

Following some observations. 

#### The diff commits from `1.0.7` to `1.3.1` can be check [here](https://github.com/bluesmoon/node-geoip/compare/v1.0.7...master)
#### Some commits can avoid this scenario/issue as follows. 
- https://github.com/bluesmoon/node-geoip/commit/8c34926eabccf06b10c884c7cd25b8098260e6e4
- https://github.com/bluesmoon/node-geoip/commit/a64195e3d5b181de149aa0921ec525dd294038fe
- https://github.com/bluesmoon/node-geoip/commit/a3fdea361f759e3ab6b4c75498e820b3aa38f972

# Verification Steps

Since the issue is intermittent then we are not able to reproduce and check it. The goal here is just to ensure that no issues will be introduced with this change. Note that it is not a major version and not break changes shows be added into them should still working fine with this version. 

This project has many tests and CI then check if it finished with success should be enough for we move forward with it.
